javascript-client/src/client.ts

```diff
@@ -140,15 +186,18 @@ export default class S3FileFieldClient {
   protected async completeUpload(
     multipartInfo: MultipartInfo, parts: UploadedPart[],
   ): Promise<void> {
-    const response = await this.api.post('upload-complete/', {
+    const response = await retry<AxiosResponse>(() => this.api.post('upload-complete/', {
       upload_signature: multipartInfo.upload_signature,
       upload_id: multipartInfo.upload_id,
       parts,
+    }), () => {
+      this.onProgress({ state: S3FileFieldProgressState.Retrying });
     });
     const { complete_url: completeUrl, body } = response.data;
 
+    // TODO support HTTP 200 error: https://github.com/girder/django-s3-file-field/issues/209
     // Send the CompleteMultipartUpload operation to S3
-    await axios.post(completeUrl, body, {
+    await retry<AxiosResponse>(() => axios.post(completeUrl, body, {
```

Comment:
I assume that if this is called repeatedly, AWS will either:
* idempotently succeed again, which is fine
* return an error in the body, which is unrecoverable (and the retries should stop)

I don't think we need to do anything more here, but we should be sure to handle this with #209.